woocommerce/google-listings-and-ads
# Add phone verification API endpoints

## Changes proposed in this Pull Request:

Closes #982.
Requires PR #998 to be finished first.

We need two new API endpoints that allow requesting a phone verification code from Google (and sending the code to the user's phone), and verifying the user-provided code.

I chose to create a new controller to handle phone verification rather than use the `ContactInformationController` because it seemed cleaner to me and there wasn't much commonality with the contact info schema.

### Change to `PhoneVerificationException`

This PR also changes `PhoneVerificationException` (introduced in #998) so it extends the pre-existing `ExceptionWithResponseData` class. My thinking was that since this `ExceptionWithResponseData` is being used in other areas to send REST error response data it makes sense to use it here also.

### Screenshots:

<img width="745" alt="Screen Shot 2021-09-10 at 11 38 36 am" src="https://user-images.githubusercontent.com/1892137/132789123-648e1821-139d-4cf0-baf2-f29bba395c72.png">

<img width="647" alt="Screen Shot 2021-09-10 at 11 47 08 am" src="https://user-images.githubusercontent.com/1892137/132789112-7234fd5b-d4c0-4c26-9ba2-e9c00986db59.png">

### Detailed test instructions:

1. Make a request to `POST /wc/gla/mc/phone-verification/request`. See `OPTIONS /wc/gla/mc/phone-verification/request` for params.
2. Save the response
3. Wait for the your verification code to come through
3. Make a request to `POST /wc/gla/mc/phone-verification/verify`. See `OPTIONS /wc/gla/mc/phone-verification/verify` for params. Use the `verification_id` value from the previous response.

### Changelog entry
